tools/otci/otci/command_handlers.py

```diff
@@ -162,6 +166,18 @@ def close(self):
     def set_line_read_callback(self, callback: Optional[Callable[[str], Any]]):
         self.__line_read_callback = callback
 
+    @classmethod
+    def set_filter(cls, filter: re.Pattern):
+        """Set a different filter for the read routine that still matches the original filter"""
+        assert filter.match('[I] ChannelMonitor: debug log')
+        assert filter.match('[I] Mac-----------: info log')
+        assert filter.match('[N] MeshForwarder-: note    log')
+        assert filter.match('[W] Notifier------: warn log')
+        assert filter.match('[C] Mle-----------: critical log')
+        assert filter.match('[-] Settings------: none log')
+        assert not filter.match('[-] Settings-----: none log')  # not enough `-` after module name
+        cls.__PATTERN_LOG_LINE = filter
```

Comment:
Why are we asserting that filter matches these very specific lines?
This is weird.

Reply:
just so the new filter will still filter the original content, but it is not a hard requirement for me

Reply:
If you think I don't need the original checks, I can remove them